src/core/automount.c

```diff
@@ -1038,18 +1037,6 @@ static int automount_dispatch_io(sd_event_source *s, int fd, uint32_t events, vo
                         goto fail;
                 }
 
-                /* Before we do anything, let's see if somebody is playing games with us? */
-                if (lstat(a->where, &st) < 0) {
-                        log_unit_warning_errno(UNIT(a), errno, "Failed to stat automount point: %m");
-                        goto fail;
-                }
-
-                if (!S_ISDIR(st.st_mode) || st.st_dev == a->dev_id) {
-                        log_unit_info(UNIT(a), "Automount point already unmounted?");
-                        automount_send_ready(a, a->expire_tokens, 0);
-                        break;
-                }
-
```

Comment:
I don't know if it will do it but `statx()` seems pretty new (the glibc doesn't provide a wrapper for it) and that would require a kernel >= 4.11, I'm not sure systemd should require such a recent kernel.

Also I'm probably missing something obvious but what's wrong leaving the stat() call to the child process ? at least PID1 is no more affected.